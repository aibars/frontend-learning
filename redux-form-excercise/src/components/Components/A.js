import React from 'react';
class A extends React.Component {

    render() {
        return (
            <a
                href={this.props.values.href}
                target="_blank"
                rel="noopener noreferrer">
                {this.props.values.label}
            </a>
        );
    }
}


export default A;
