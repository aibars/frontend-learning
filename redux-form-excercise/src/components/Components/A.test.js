import React from 'react';
import { shallow } from 'enzyme';

import A from './A';

describe('A', () => {
  it('works ok', () => {
    const component = shallow((
      <A
        values={{
          href: 'http://www.google.com',
          label: 'Click me!',
        }}
      />
    ));

    expect(component).toMatchSnapshot();
  });
});
