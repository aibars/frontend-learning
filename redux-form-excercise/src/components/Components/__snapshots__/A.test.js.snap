// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A works ok 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <A
    values={
      Object {
        "href": "http://www.google.com",
        "label": "Click me!",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Click me!",
      "href": "http://www.google.com",
      "rel": "noopener noreferrer",
      "target": "_blank",
    },
    "ref": null,
    "rendered": "Click me!",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click me!",
        "href": "http://www.google.com",
        "rel": "noopener noreferrer",
        "target": "_blank",
      },
      "ref": null,
      "rendered": "Click me!",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
